celer/tests/test_lasso.py

```diff
@@ -210,7 +210,16 @@ def test_weights_lasso():
 
     assert_allclose(clf1.coef_, clf2.coef_ / weights)
 
+    # support inf weights
+    weights = np.ones(X.shape[1])
+    weights[0] = np.inf
+    alpha_max = norm(X.T.dot(y) / weights, ord=np.inf) / X.shape[0]
+
+    clf1 = Lasso(alpha=alpha_max, weights=weights,
+                 max_iter=1, max_epochs=20).fit(X, y)
+
```

Comment:
you need to assert something after the test. I would use a lower alpha, and assert that the solver converges + all coefficients associated to infinite weights are 0